From 4b66c851081eb5835ce9e745a6fd7213d31df2a3 Mon Sep 17 00:00:00 2001
From: Rentib <sbitner420@tutanota.com>
Date: Wed, 18 Sep 2024 15:13:15 +0200
Subject: [PATCH] hide vacant tags

---
 dam.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/dam.c b/dam.c
index dfcf9e6..b80afe8 100644
--- a/dam.c
+++ b/dam.c
@@ -161,8 +161,6 @@ bar_draw(Bar *bar)
 	int i;
 	int tw = 0;
 	int x = 0, w;
-	int boxs = bar->drw->font->height / 9;
-	int boxw = bar->drw->font->height / 6 + 2;
 	DrwBuf *buf;
 
 	if (!bar->configured)
@@ -181,12 +179,11 @@ bar_draw(Bar *bar)
 	}
 
 	for (i = 0; i < LENGTH(tags); i++) {
+		if (!((bar->mtags | bar->ctags | bar->urg) & 1 << i))
+			continue;
 		w = TEXTW(bar, tags[i]) + tagspadding;
 		drwl_setscheme(bar->drw, colors[bar->mtags & 1 << i ? SchemeSel : SchemeNorm]);
 		drwl_text(bar->drw, x, 0, w, bar->height, bar->lrpad / 2 + tagspadding / 2, tags[i], bar->urg & 1 << i);
-		if (bar->ctags & 1 << i)
-			drwl_rect(bar->drw, x + boxs, boxs, boxw, boxw, 0,
-				bar->urg & 1 << i);
 		x += w;
 	}
 
@@ -451,9 +448,14 @@ pointer_handle_frame(void *data, struct wl_pointer *wl_pointer)
 
 	lw = TEXTW(selbar, selbar->layout);
 
-	do
-		x += TEXTW(selbar, tags[i]) + tagspadding;
-	while (pointer.x >= x && ++i < LENGTH(tags));
+	unsigned occ = 0;
+	Bar *bar;
+	wl_list_for_each(bar, &bars, link)
+		occ |= bar->mtags | bar->ctags | bar->urg;
+	do {
+		if (occ & 1 << i)
+			x += TEXTW(selbar, tags[i]) + tagspadding;
+	} while (pointer.x >= x && ++i < LENGTH(tags));
 
 	if (i < LENGTH(tags))
 		click = ClkTagBar;
-- 
2.46.1

